app_test.go

```diff
@@ -1124,7 +1125,10 @@ func Test_Test_Timeout(t *testing.T) {
 
 	app.Get("/", testEmptyHandler)
 
-	resp, err := app.Test(httptest.NewRequest(MethodGet, "/", nil), -1)
+	resp, err := app.Test(httptest.NewRequest(MethodGet, "/", nil), TestConfig{
+		Timeout:       -1,
+		FailOnTimeout: false,
```

Comment:
@grivera64 if failOnTimeout is false, do we really need a -1 for the timeout? couldn't we do this implicitly?

Reply:
@ReneWerner87 `FailOnTimeout` does not directly relate to the `Timeout` field, but the naming may mislead users to believe so.

When true, `FailOnTimeout` tells the test to return an error after timing out. If `FailOnTimeout` is false, the test will return the most up-to-date response before timing out without returning an error. If there is no timeout (Timeout: -1), then the value of `FailOnTimeout` doesn't affect the test. For clarity, I kept `FailOnTimeout` as false since there is no timeout that can lead to a failing test in the first place.

Maybe a less misleading name for the field would be something like `IgnoreResponseOnTimeout` or something along those lines.

What are your thoughts about this?

Reply:
Actually, I think I misunderstood your initial comment, my mistake. With the current setup, we could just implicitly do the reverse: If there is no timeout (0 seconds), then FailOnTimeout isn't really required, since the code that checks the value of FailOnTimeout wouldn't be checked in that case.

I've updated the middleware testing code to implicitly use `FailOnTimeout: false` when `Timeout: 0`.